### PR TITLE
fix: Add a check in PR's `requested_reviewers`

### LINKED ISSUE #206: Action does not perform review when requesting for re-review
**Steps to reproduce:**
1. Setup action, create a PR, and have it auto reviewed
2. Request for **re-review** in PR, **without** further code changes, review status should be something like this
image
3. Trigger `auto-approve-action`

**Actual output:**
Does not do a re-review, but logs a warning: `Current user already approved pull request #31, nothing to do`, despite the PR needing a re-review

**Expected output:** 
`auto-approve-action` should perform a re-review as the status is "Awaiting requested review...", if re-review is successful, status should be updated to "Approved"

Here's a sample PR: https://github.com/vincejv/fpi-framework/pull/31
image

### PR DESCRIPTION
**Pull request by vincejv**
Add a check in PR's `requested_reviewers` before considering the PR as already approved, fixes the issue wherein action skips approving a PR that was approved before but requested a re-review, closes #206

### CODE REVIEW

File: src/approve.ts [hunk 1]
```diff
@@ -52,7 +52,13 @@ export async function approve(
...
         review.user?.login == login &&
         review.commit_id == commit &&
-        review.state == "APPROVED"
+        review.state == "APPROVED" &&
+        ((
+          pull_request.data.requested_reviewers != undefined &&
+          pull_request.data.requested_reviewers.filter(reviewer => reviewer.login == login).length == 0
+        ) ||
+        pull_request.data.requested_reviewers == undefined
+        )
```
Comment (hmarr) [inline src/approve.ts]: This `if` statement is getting a little hard to follow 😅 

If we use the [optional chaining operator (`?.`)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) we can avoid both of the checks against `undefined` and remove the additional `||`, which simplifies it a bit.

Additionally, [`!Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a slightly more concise way of expressing `filter(...).length == 0`.

```suggestion
        !pull_request.data.requested_reviewers?.some(
          (user) => user.login === login
        )
```

**Review by hmarr — COMMENTED** [review]
Thanks for submitting this!

I've added one comment on the big if statement, but this looks great apart from that.

Also, the linter seems to be failing in CI, so can you run `npm run format` on your branch?

File: src/approve.ts (hunk 1 above)
Comment (vincejv) [reply src/approve.ts]: Hi @hmarr, corrected the format and used your suggestions, can you help to check? 😊

**Review by hmarr — APPROVED** [review]
Looking good! I'll merge and release tomorrow 🚀

**Comment by hmarr** [comment]
Released as [v3.1.0](https://github.com/hmarr/auto-approve-action/releases/tag/v3.1.0) 🚀 

Thanks for contributing!
